[BUG] Pytest failures: test_structural_sw_rw TestSynth OBI

- [ X ] I have verified this bug has not already been reported on the [upstream issue tracker](https://github.com/SystemRDL/PeakRDL-regblock/issues). Upstream bugfixes/features are merged into this repository with each upstream release.
- [ X ] I have reviewed this project's [contribution guidelines](https://github.com/SystemRDL/PeakRDL-regblock-vhdl/blob/main/CONTRIBUTING.md)

**Describe the bug**
I noticed that the OBI interface synthesis tests are failing on my build environment. This is not the case for the flattened OBI interface or in the upstream (PeakRDL-regblock) repo.

Upon closer inspection, the "aid" and "rid" record elements are not constrained in the vhdl_regblock s_obi_i/o port declarations, which Vivado is not happy about (ERROR: [Synth 8-318] illegal unconstrained array declaration 's_obi_i').

Is this expected behavior or unique to my build environment? If not, the next question would be how the ID width should be constrained. Possibly with a port generic, or by synthesizing the adapter instead?

**Expected behavior**
Pytests passing

**Details**
Python 3.11
Vivado 2025.2 (same result with 2024.1)
`pytest --sim-tool questa --synth-tool vivado

...

----------------------- Captured stderr call --------------------

ERROR: [Synth 8-318] illegal unconstrained array declaration 's_obi_i' [/PeakRDL-regblock-vhdl/tests/test_structural_sw_rw/run.out/TestSynth_OBI_False_False_False/vhdl_regblock.vhd:16]
ERROR: [Synth 8-285] failed synthesizing module 'vhdl_regblock' [/PeakRDL-regblock-vhdl/tests/test_structural_sw_rw/run.out/TestSynth_OBI_False_False_False/vhdl_regblock.vhd:29]
ERROR: [Common 17-69] Command failed: Synthesis failed - please see the console or run log file for details

========== short test summary info ===============

FAILED test_structural_sw_rw/testcase.py::TestSynth_OBI_True_True_True::test_dut - subprocess.CalledProcessError: Command '['vivado', '-nojournal', '-notrace', '-mode', 'batch', '-log', 'out.log', '-source',
FAILED test_structural_sw_rw/testcase.py::TestSynth_OBI_True_True_False::test_dut - subprocess.CalledProcessError: Command '['vivado', '-nojournal', '-notrace', '-mode', 'batch', '-log', 'out.log', '-source',
FAILED test_structural_sw_rw/testcase.py::TestSynth_OBI_True_False_True::test_dut - subprocess.CalledProcessError: Command '['vivado', '-nojournal', '-notrace', '-mode', 'batch', '-log', 'out.log', '-source', 
FAILED test_structural_sw_rw/testcase.py::TestSynth_OBI_True_False_False::test_dut - subprocess.CalledProcessError: Command '['vivado', '-nojournal', '-notrace', '-mode', 'batch', '-log', 'out.log', '-source',
FAILED test_structural_sw_rw/testcase.py::TestSynth_OBI_False_True_True::test_dut - subprocess.CalledProcessError: Command '['vivado', '-nojournal', '-notrace', '-mode', 'batch', '-log', 'out.log', '-source',
FAILED test_structural_sw_rw/testcase.py::TestSynth_OBI_False_True_False::test_dut - subprocess.CalledProcessError: Command '['vivado', '-nojournal', '-notrace', '-mode', 'batch', '-log', 'out.log', '-source',
FAILED test_structural_sw_rw/testcase.py::TestSynth_OBI_False_False_True::test_dut - subprocess.CalledProcessError: Command '['vivado', '-nojournal', '-notrace', '-mode', 'batch', '-log', 'out.log', '-source',
FAILED test_structural_sw_rw/testcase.py::TestSynth_OBI_False_False_False::test_dut - subprocess.CalledProcessError: Command '['vivado', '-nojournal', '-notrace', '-mode', 'batch', '-log', 'out.log', '-source',

========= 8 failed, 391 passed in 8575.01s (2:22:55) ============
`

--- a/src/peakrdl_regblock_vhdl/cpuif/obi/__init__.py
+++ b/src/peakrdl_regblock_vhdl/cpuif/obi/__init__.py
@@ -16,10 +16,12 @@ class OBI_Cpuif(CpuifBase):
             "s_obi_i : in obi_subordinate_in_intf(",
            f"    addr({self.addr_width-1} downto 0),",
            f"    be({self.data_width_bytes-1} downto 0),",
-           f"    wdata({self.data_width-1} downto 0)",
+           f"    wdata({self.data_width-1} downto 0),",
+            "    aid(ID_WIDTH-1 downto 0)",
             ");",
             "s_obi_o : out obi_subordinate_out_intf(",
-           f"   rdata({self.data_width-1} downto 0)",
+           f"    rdata({self.data_width-1} downto 0),",
+            "    rid(ID_WIDTH-1 downto 0)",
             ")",
         ])
 
@@ -56,6 +58,10 @@ class OBI_Cpuif(CpuifBase):
     def id_width(self) -> int:
         return 1  # Default ID width
 
+    @property
+    def parameters(self) -> List[str]:
+        return [f"ID_WIDTH : positive := {self.id_width}"]
+
 
 class OBI_Cpuif_flattened(OBI_Cpuif):
     is_interface = False
@@ -87,7 +93,3 @@ class OBI_Cpuif_flattened(OBI_Cpuif):
 
     def signal(self, name: str) -> str:
         return "s_obi_" + name
-
-    @property
-    def parameters(self) -> List[str]:
-        return ["ID_WIDTH : positive := 1"]

--- a/src/peakrdl_regblock_vhdl/field_logic/sw_onread.py
+++ b/src/peakrdl_regblock_vhdl/field_logic/sw_onread.py
@@ -1,6 +1,6 @@
-from typing import TYPE_CHECKING, List
+from typing import TYPE_CHECKING, List, Optional
 
-from systemrdl.rdltypes import OnReadType
+from systemrdl.rdltypes.builtin_enums import OnReadType
 
 from .bases import NextStateConditional
 from ..vhdl_int import VhdlInt
@@ -9,7 +9,7 @@ if TYPE_CHECKING:
     from systemrdl.node import FieldNode
 
 class _OnRead(NextStateConditional):
-    onreadtype = None # type: OnReadType
+    onreadtype = None # type: Optional[OnReadType]
     def is_match(self, field: 'FieldNode') -> bool:
         return field.get_property('onread') == self.onreadtype
 

